tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/RequestValidationHelperTests.cs

```diff
@@ -65,43 +66,33 @@ public void ClientAssertionRequestValidatorMismatchParameterTest()
                 ValidTo = ConvertToTimeT(DateTime.UtcNow + TimeSpan.FromSeconds(JwtToAadLifetimeInSeconds))
             };
 
-            var parameters = new AuthenticationRequestParameters
-            {
-                ClientCredential = credential,
-                SendCertificate = false,
-                Authority = Authority.CreateAuthority(
-                    _serviceBundle,
-                    MsalTestConstants.AuthorityCommonTenant),
-                Endpoints = new AuthorityEndpoints(null, null, Audience1)
-            };
-
             // Validate cached client assertion with parameters
-            Assert.IsTrue(RequestValidationHelper.ValidateClientAssertion(parameters));
+            Assert.IsTrue(ClientCredentialHelper.ValidateClientAssertion(credential, new AuthorityEndpoints(null, null, Audience1), false));
 
             // Different audience
             credential.Audience = Audience2;
 
             // cached assertion should be invalid
-            Assert.IsFalse(RequestValidationHelper.ValidateClientAssertion(parameters));
+            Assert.IsFalse(ClientCredentialHelper.ValidateClientAssertion(credential, new AuthorityEndpoints(null, null, Audience1), false));
```

Comment:
> Audience1 [](start = 120, length = 10)

should this be Audience2?

Reply:
from the code it looks like it should be but it makes the test fail.  so i will need to look into this.

---
In reply to: [248927197](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/798#discussion_r248927197) [](ancestors = 248927197)